luigi/process.py

```diff
@@ -97,11 +99,16 @@ def daemonize(cmd, pidfile=None, logdir=None, api_port=8082, address=None, unix_
     stdout_proxy = open(stdout_path, 'a+')
     stderr_proxy = open(stderr_path, 'a+')
 
-    ctx = daemon.DaemonContext(
-        stdout=stdout_proxy,
-        stderr=stderr_proxy,
-        working_directory='.'
-    )
+    daemon_kwargs = {
+        "stdout": stdout_proxy,
+        "stderr": stderr_proxy,
+        "working_directory": '.',
+    }
+
+    if (StrictVersion(gdvi().get("version") or "0.0.0") >= StrictVersion("2.1.0")):
```

Comment:
i also would prefer it if you just apply "it's easier to ask for forgiveness than permission" and just did two calls – try first with the argument, if it throws an `ArgumentError` then try without